### DISCUSSION TITLE
List pull requests using search query and date does not include few results.

### DISCUSSION THREAD

**Discussion by prudhvigodithi** [community/community#46796] [Pull Requests]
### Select Topic Area

Bug

### Body

I’m trying to get PR’s created which are older than 3 months (the 3 months old date to current date is `2022-11-08`)

Total PR’s 3 months older: `1594 (1586 closed + 8 open)`
https://github.com/opensearch-project/OpenSearch-Dashboards/pulls?q=is%3Apr+created%3A%3C%3D2022-11-08+sort%3Acreated-asc

The total count should be `1599`

The missing issue from GH search which are also older than `2022-11-08` are
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/85
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/237
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/484
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/1650
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/610

To reproduce:

API https://api.github.com/search/issues?state=all&q=repo:opensearch-project/OpenSearch-Dashboards+is:pr+created:%3C=2022-11-08&per_page=33 shows as `1594` should be `1599`

(or)

https://github.com/opensearch-project/OpenSearch-Dashboards/pulls?q=is%3Apr+created%3A%3C%3D2022-11-08

**Comment by peterzhuamazon** [comment]
I verified this and can confirm it has exactly the same output as @prudhvigodithi pointed out.
Tho I am not seeing any patterns among the above 6 missed issues/prs.
Not sure if this is an edge case. Thanks.

**Comment by prudhvigodithi** [comment]
Found one more data point with PR's API
The total PR's shows as `1798` (1787+11) https://github.com/opensearch-project/opensearch-build/pulls?q=is%3Apr++sort%3Acreated-asc+ 
![Screenshot 2023-02-09 at 1 32 15 PM](https://user-images.githubusercontent.com/28733332/217943883-ce7060bf-7665-425f-9907-7e4b666b5f6b.png)
From the API https://api.github.com/search/issues?state=all&q=repo:opensearch-project/opensearch-build+is:pr+created:%3E=2019-01-01 it shows as 1796 even though the start date is `created:>=2019-01-01` whereas the repo oldest PR is at Apr 20, 2021
![Screenshot 2023-02-09 at 1 33 13 PM](https://user-images.githubusercontent.com/28733332/217944057-3d290036-f728-4a60-a318-41dba5587a19.png)
@dblock @bbarani
